src/extension.ts

```diff
@@ -28,9 +28,19 @@ export function activate(context: ExtensionContext): void {
             // dirty way to get decorations work after showing disassembly
             setTimeout(() => decorator.updateSelection(srcEditor), 500);
 
-            workspace.onDidCloseTextDocument((d: TextDocument) => {
-                if (d.uri == srcEditor.document.uri) provider.unload(d.uri);
+            // keep reference to documents since editors can change under us
+            const srcDocument = srcEditor.document;
+            const asmDocument = asmEditor.document;
+
+            const disposable = workspace.onDidCloseTextDocument((d: TextDocument) => {
+                if (d === srcDocument || d === asmDocument) {
```

Comment:
I'm able to test the case of `disposable` being invoked after closing `asmDocument`. This is not related to your patch, but in general even without your patch. Were you able repro the case of disposable being invoked after closing the asmDocument?

Reply:
Yea, great question, I noticed that too. Although, sometime later after closing the asm tab, `onDidCloseTextDocument` would get called for the `asmDocument` (typically after `c_cpp_properties.json` would get released as well, so VS Code seems to keep it cached for some time). I figured I'd leave that in for completeness/correctness anyways since it does eventually get called, at least sometimes. ¯\_(ツ)_/¯

Reply:
Right. I encountered the same behaviour too. `leave that in for completeness/correctness ` -> +1